File: apps/sqlalchemy-example/app/sources/externalTopics.ts
/**
 * External Kafka Topics (CDC Events from PostgreSQL)
 */

import { ExternalStream } from '@514labs/moose-lib';
import { RedpandaPgCdcPayload } from '../models';

/**
 * Redpanda Connect streams CDC events from PostgreSQL to this topic
 * Topic: sqlalchemy_cdc_events
 */
export const SqlAlchemyCdcEventsStream = new ExternalStream<RedpandaPgCdcPayload<any>>(
  'sqlalchemy_cdc_events',
  {
    sourceTopic: 'sqlalchemy_cdc_events',
  }
);

File: apps/sqlalchemy-example/app/sources/external_topics.py
"""
External Kafka Topics (CDC Events from PostgreSQL)

Redpanda Connect streams CDC events from PostgreSQL to these topics.
Topic: sqlalchemy_cdc_events
"""

from moose_lib import Stream, LifeCycle
from app.models import RedpandaPgCdcPayload


# External CDC event stream from Redpanda Connect
SqlAlchemyCdcEventsStream = Stream[RedpandaPgCdcPayload](
    name='sqlalchemy_cdc_events',
    lifecycle=LifeCycle.EXTERNALLY_MANAGED
)